### PR TITLE
Ocd 2474 - Update invalid input fields

### PR DESCRIPTION
**Pull request by brianlindsy**

### CODE REVIEW

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/listing/reviewer/UnattestedCriteriaWithDataReviewer.java [hunk 1]
```diff
@@ -20,75 +22,58 @@ public void review(CertifiedProductSearchDetails listing) {
...
                 if (listing.getSed() != null && listing.getSed().getTestTasks() != null
                         && listing.getSed().getTestTasks().size() > 0) {
                     for (TestTask tt : listing.getSed().getTestTasks()) {
+                        ArrayList<CertificationCriterion> remove = new ArrayList<CertificationCriterion>();
                         for (CertificationCriterion ttCriteria : tt.getCriteria()) {
                             if (ttCriteria.getNumber() != null && ttCriteria.getNumber().equals(cert.getNumber())) {
-                                listing.getErrorMessages().add(msgUtil.getMessage(
-                                        "listing.criteria.falseCriteriaHasData",
-                                        cert.getNumber(), "Test Tasks"));
+                                remove.add(ttCriteria);
```
Comment (brianlindsy) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/listing/reviewer/UnattestedCriteriaWithDataReviewer.java]: I think this is the correct logic for this, can someone confirm?

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/listing/reviewer/UnattestedCriteriaWithDataReviewer.java [hunk 2]
```diff
@@ -20,75 +22,58 @@ public void review(CertifiedProductSearchDetails listing) {
...
                 if (listing.getSed() != null && listing.getSed().getUcdProcesses() != null
                         && listing.getSed().getUcdProcesses().size() > 0) {
                     for (UcdProcess ucd : listing.getSed().getUcdProcesses()) {
+                        ArrayList<CertificationCriterion> remove = new ArrayList<CertificationCriterion>();
                         for (CertificationCriterion ucdCriteria : ucd.getCriteria()) {
                             if (ucdCriteria.getNumber() != null && ucdCriteria.getNumber().equals(cert.getNumber())) {
-                                listing.getErrorMessages().add(msgUtil.getMessage(
-                                        "listing.criteria.falseCriteriaHasData",
-                                        cert.getNumber(), "UCD Processes"));
+                                remove.add(ucdCriteria);
```
Comment (brianlindsy) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/listing/reviewer/UnattestedCriteriaWithDataReviewer.java]: I think this is the correct logic for this, can someone confirm?

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/listing/reviewer/UnattestedCriteriaWithDataReviewer.java (hunk 1 above)
Comment (andlar) [reply chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/listing/reviewer/UnattestedCriteriaWithDataReviewer.java]: I think that looks okay. Did you test it in the real world?

**Comment by brianlindsy** [comment]
I think I still need all of the errors, because pending validator uses them too

**Comment by andlar** [comment]
> I think I still need all of the errors in the properties file because pending validator uses them too

Works for me. Are the RELEASE NOTES correct?

**Pull request merged**